src/kv1/read.rs

```diff
@@ -14,20 +14,99 @@
 // along with this program.  If not, see
 // <http://www.gnu.org/licenses/>.
 
+use super::de_from_date_string;
+use crate::collection::CollectionWithId;
+use crate::common_format::CalendarDate;
 use crate::model::Collections;
+use crate::objects::{Calendar, Date, ExceptionType};
 use crate::read_utils::FileHandler;
 use crate::Result;
+use csv;
+use failure::bail;
 use log::info;
+use serde_derive::Deserialize;
+use std::collections::BTreeSet;
+
+#[derive(Deserialize, Debug)]
+struct OPerDay {
+    #[serde(rename = "[Recordtype]")]
+    record_type: String,
+    #[serde(rename = "[Version number]")]
+    version_number: String,
+    #[serde(rename = "[Implicit/Explicit]")]
+    implicit_explicit: String,
+    #[serde(rename = "[DataOwnerCode]")]
+    data_owner_code: String,
+    #[serde(rename = "[OrganizationalUnitCode]")]
+    org_unit_code: String,
+    #[serde(rename = "[ScheduleCode]")]
+    schedule_code: String,
+    #[serde(rename = "[ScheduleTypeCode]")]
+    schedule_type_code: String,
+    #[serde(rename = "[ValidDate]", deserialize_with = "de_from_date_string")]
+    valid_date: Date,
+    #[serde(rename = "[Description]")]
+    description: String,
+}
 
 /// Generates calendars
 pub fn read_operday<H>(file_handler: &mut H, collections: &mut Collections) -> Result<()>
 where
     for<'a> &'a mut H: FileHandler,
 {
-    info!("Reading OPERDAYXXX.TMI");
-
-    // collections.calendars = CollectionWithId::new(calendars)?;
-
+    let file = "OPERDAYXXX.TMI";
+    let (reader, _) = file_handler.get_file_if_exists(file)?;
```

Comment:
`let (reader, path) = file_handler.get_file_if_exists(file)?;`

Reply:
path not used, so intentionally deleted